src/frontend/src/flows/addDevice/welcomeView/showVerificationCode.ts

```diff
@@ -65,79 +85,114 @@ export const showVerificationCode = async (
   tentativeRegistrationInfo: TentativeRegistrationInfo,
   credentialToBeVerified: CredentialId
 ): Promise<void> => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-  render(pageContent(userNumber, alias, tentativeRegistrationInfo), container);
-  return init(
+  const countdown = AsyncCountdown.fromNanos(
+    tentativeRegistrationInfo.device_registration_timeout
+  );
+
+  showVerificationCodePage({
+    userNumber,
+    alias,
+    tentativeRegistrationInfo,
+    remaining: countdown.remainingFormattedAsync(),
+    cancel: () => {
+      countdown.stop();
+      window.location.reload();
+    },
+  });
+
+  const pollResult = await poll({
     userNumber,
     connection,
-    tentativeRegistrationInfo.device_registration_timeout,
-    credentialToBeVerified
-  );
+    credentialToBeVerified,
+    shouldStop: () => countdown.hasStopped(),
+  });
+
+  return await handlePollResult(userNumber, pollResult);
```

Comment:
nth: there is a particular reason why this function does not take an object as argument while other functions of this PR do?

Reply:
Nope! Will change, good point